### PR TITLE
[workflows] Add a new workflow for checking commit access qualifications

### PR DESCRIPTION
**Pull request by tstellar**
This workflow produces a list of users that should be moved to the LLVM Triagers team.  An admin will review the list and then manually move people to the LLVM Triagers Team.

I've also added some documentation to the Developer Policy about the Triage Role.

See https://discourse.llvm.org/t/rfc2-new-criteria-for-commit-access/77110 for more details.

### CODE REVIEW

File: llvm/docs/DeveloperPolicy.rst
Comment (igorkudrin) [inline llvm/docs/DeveloperPolicy.rst:540]: This reads like a user should have at least 5 interactions _per month_ to keep their commit access.

**Comment by tstellar** [comment]
cc @lattner For the DeveloperPolicy update.

**Comment by tstellar** [comment]
Ping.

File: llvm/docs/DeveloperPolicy.rst
Comment (boomanaiden154) [inline llvm/docs/DeveloperPolicy.rst:530]: It seems like this should be probably be reviewed in a separate patch/maybe accompanied by a post on Discourse given it's more of a project policy thing? At least the second sentence of the first paragraph.

The second paragraph can probably just be landed, but probably in a separate commit to keep things cleanish.

File: .github/workflows/commit-access-review.py
Comment (boomanaiden154) [inline .github/workflows/commit-access-review.py:1]: We should probably have the LLVM copyright notice at the top of the file. See something like https://github.com/llvm/llvm-project/blob/b330d800cb7917e537b05a23febfe188401c5628/llvm/utils/mlgo-utils/mlgo/corpus/combine_training_corpus.py#L1.

We should probably standardize the python header comment block at some point, but that's a separate issue...

File: .github/workflows/commit-access-review.py
Comment (boomanaiden154) [inline .github/workflows/commit-access-review.py:435]: Put this into a `with` statement to avoid manually having to call the `f.close()`? Also, a more descriptive name would probably be useful.

File: .github/workflows/commit-access-review.py
Comment (boomanaiden154) [inline .github/workflows/commit-access-review.py:132]: This seems left over from testing?

File: .github/workflows/commit-access-review.py
Comment (boomanaiden154) [inline .github/workflows/commit-access-review.py:391]: The line break here seems misplaced/inconsistent with the rest of the file?

**Review by boomanaiden154 — COMMENTED** [review]
Mostly looks good to me, a couple nits.

A couple high level comments
* The Github library we're using doesn't support querying the GraphQL API? And there's no other well-supported libraries that do? It would be nice if we could do that instead.
* It would be nice if we could unit test this. That would probably require mocking the whole thing though and require a lot of work. I'm not going to block on that because that would take quite a bit of effort, the results are going to be manually curated anyways, and something here is better than nothing, but it would be nice to have.
* Typing annotations/doc strings - Would be nice to have, but not going to require them. The function names/comments make things pretty clear already.

One main question:
* Do you have numbers on how many API requests this uses? At a quick glance I would expect it to use quite a bit. If it's under the `RELEASE_TASKS_USER_TOKEN`, it might be fine though?

**Comment by tstellar** [comment]
> * Do you have numbers on how many API requests this uses? At a quick glance I would expect it to use quite a bit. If it's under the `RELEASE_TASKS_USER_TOKEN`, it might be fine though?

It uses between 3,000 and 4,000 API calls typically, depending on how many users there are.

File: llvm/docs/DeveloperPolicy.rst
Comment (tstellar) [reply llvm/docs/DeveloperPolicy.rst:530]: OK, I'll do a separate PR with the doc changes.

**Comment by tstellar** [comment]
> * The Github library we're using doesn't support querying the GraphQL API? And there's no other well-supported libraries that do? It would be nice if we could do that instead.
> 

I could not find one that works with GitHub.

> * It would be nice if we could unit test this. That would probably require mocking the whole thing though and require a lot of work. I'm not going to block on that because that would take quite a bit of effort, the results are going to be manually curated anyways, and something here is better than nothing, but it would be nice to have.
> 

I think if the whole process were automated, then this would be more important, but the manual review means mistakes can get caught and have a smaller impact.

> * Typing annotations/doc strings - Would be nice to have, but not going to require them. The function names/comments make things pretty clear already.

I've added these.

**Comment by boomanaiden154** [comment]
> I could not find one that works with GitHub.

Ack. I guess if we need this functionality anywhere else

> I think if the whole process were automated, then this would be more important, but the manual review means mistakes can get caught and have a smaller impact.

Definitely agree.

> I've added these.

Thanks. I definitely think it makes things a bit easier to follow.

**Review by boomanaiden154 — APPROVED** [review]
Seems reasonable enough to me.

**Pull request merged**
